orange/plugins/jwt_auth/handler.lua

```diff
@@ -0,0 +1,167 @@
+local ipairs = ipairs
+local type = type
+local string_find = string.find
+
+local jwt = require("resty.jwt")
+local utils = require("orange.utils.utils")
+local cjson = require("cjson")
+local orange_db = require("orange.store.orange_db")
+local judge_util = require("orange.utils.judge")
+local handle_util = require("orange.utils.handle")
+local BasePlugin = require("orange.plugins.base_handler")
+
+string.split = function(s, p)
+    local rt= {}
+    string.gsub(s, '[^'..p..']+', function(w) table.insert(rt, w) end )
+    return rt
+end
+
+local function jwt_auth(credentials, jwttoken)
+    local secret = credentials.secret
+    local jwtobj = jwt:verify(secret, jwttoken)
+    if jwtobj["verified"] then
+        for i, payload in ipairs(credentials.payload) do  
+            if payload.type == 1 then
+                ngx.req.set_header(payload.target_key, jwtobj.payload[payload.key])
+            end
+        end 
+
+        return true
+    end
+    return false
+end
+
+local function is_credential_in_header(credentials, headers)
+    local jwttoken = headers["Authorization"]
+    if not jwttoken then return false end
+    return jwt_auth(credentials, string.split(jwttoken, " ")[2])
+end
+
+
+local function is_credential_in_query(credentials, query)
+    if not query or not "token" then return false end
+    local jwttoken = query["token"]
+    return jwt_auth(credentials, jwttoken)
+end
+
+
+
+local function is_authorized(credentials, headers, query)
+    if not credentials then return false end
+    local authorized = false
+
+    if is_credential_in_header(credentials,headers) then
+                authorized = true
+    elseif is_credential_in_query(credentials,query) then
+                authorized = true
+    end
+
+    return authorized
+end
+
+local function filter_rules(sid, plugin, ngx_var_uri, headers, query)
+    local rules = orange_db.get_json(plugin .. ".selector." .. sid .. ".rules")
+    if not rules or type(rules) ~= "table" or #rules <= 0 then
+        return false
+    end
+
+    for i, rule in ipairs(rules) do
+        local enable = rule.enable
+        if enable == true then
+            -- judge阶段
+            local pass = judge_util.judge_rule(rule, plugin)
+
+            -- handle阶段
+            if pass then
+                local handle = rule.handle
+                if handle.credentials then
+                    if handle.log == true then
+                        ngx.log(ngx.INFO, "[JwtAuth-Pass-Rule] ", rule.name, " uri:", ngx_var_uri)
```

Comment:
looks should move this 3 lines in the 'if authorized then' block, and add another 'JwtAuth-Forbidden-Rule' log in the 'else' block.